### PR TITLE
implement the basic room list function

### PR DESCRIPTION
**Pull request by mingxin-yu**
Achieved the basic functionality of the room component. Might need future improvement on styling and routing.

### CODE REVIEW

File: src/app/rooms/rooms.component.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Component, OnInit } from '@angular/core';
+import { Room } from '../room';
+import { RoomService } from '../room.service';
+
```
Comment (ghost) [inline src/app/rooms/rooms.component.ts:4]: Add comment explaining what this component is

File: src/app/rooms/rooms.component.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<h2>All Rooms</h2>
+<ul class="rooms">
+  <li *ngFor="let room of rooms">
+    <a routerLink="/detail/{{room.id}}">
```
Comment (ghost) [inline src/app/rooms/rooms.component.html:4]: Does this router work?

Comment (mingxin-yu) [reply src/app/rooms/rooms.component.html:4]: No. It does not work. But I need this for future testing on the room details component. I don't want to come back to room component to add this line of code for future testing.

Comment (ghost) [reply src/app/rooms/rooms.component.html:4]: If this doesn't do anything, this shouldn't be here.

Comment (mingxin-yu) [reply src/app/rooms/rooms.component.html:4]: Okay

Comment (mingxin-yu) [reply src/app/rooms/rooms.component.html:4]: Yes, it does navigate you to that pages except for the fact that the page does not show anything and creates an error message in the console.

File: src/app/rooms/rooms.component.ts (hunk 1 above)
Comment (mingxin-yu) [reply src/app/rooms/rooms.component.ts:4]: Ok. Done.

File: src/app/rooms/rooms.component.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * This component gives you a overview of each room with it's id.
```
Comment (ghost) [inline src/app/rooms/rooms.component.ts]: dont use word like "use, I". This is describing the function of a component, not telling a story.
Just say something like, displays a list of rooms with its overview or something.

**Review by ghost — APPROVED** [review]

File: src/app/rooms/rooms.component.ts (hunk 3 above)
Comment (mingxin-yu) [reply src/app/rooms/rooms.component.ts]: Changed. See new update

**Pull request merged**